test/process_test.cc

```diff
@@ -32,12 +33,34 @@ class ProcessTest : public TestWithParam<Envoy::Network::Address::IpVersion> {
       : loopback_address_(Envoy::Network::Test::getLoopbackAddressUrlString(GetParam())),
         options_(TestUtility::createOptionsImpl(
             fmt::format("foo --duration 1 -v error --rps 10 https://{}/", loopback_address_))){};
-  void runProcess(RunExpectation expectation) {
+  void runProcess(RunExpectation expectation, bool do_cancel = false) {
     ProcessPtr process = std::make_unique<ProcessImpl>(*options_, time_system_);
     OutputCollectorImpl collector(time_system_, *options_);
+    std::thread cancel_thread;
+    if (do_cancel) {
+      cancel_thread = std::thread([&process] {
+        sleep(5);
```

Comment:
Would it be worth also testing the case where the sigterm comes before the threads started?

Reply:
I agree that would be a good thing to have. But it seems challenging to add such a test (without adding code to the implementation to support it). I need to think about it some more.

Reply:
I think I misunderstood, added a test in 20e610867dbbbeb48d07952fa0de45b1bad62d52

Reply:
Whoops @eric846 , my comment crossed yours :-)

Reply:
@eric846 IMHO your suggestion did yield a meaning full test which turned out easy to add. So thanks!

Still, thinking more about it, I feel this isn't complete. A repeated and non deterministic test which fuzzes timing to hopefully strike all possible flows with regard to cancellation might be good to have too at some point. But this isn't trivial to add today.